Subject: [PATCH] Wi-Fi : Abort setWifiEnabled if WifiController not ready

While executing setWifiEnabled, it send commands to WifiController.
If WifiController is not started by this time, StateMachine reports
crash which creates device reboot.
Thus, check if WiFiController is started before invoking any
command to the same. Indicate a failure in such an instance.

diff --git a/service/java/com/android/server/wifi/WifiServiceImpl.java b/service/java/com/android/server/wifi/WifiServiceImpl.java
index a5e5c3d..5feee89 100644
--- a/service/java/com/android/server/wifi/WifiServiceImpl.java
+++ b/service/java/com/android/server/wifi/WifiServiceImpl.java
@@ -188,6 +188,7 @@
     private WifiScanner mWifiScanner;
     private WifiLog mLog;
 
+    private boolean mIsControllerStarted = false;
     /**
      * Asynchronous channel to WifiStateMachine
      */
@@ -555,6 +556,7 @@
             Log.wtf(TAG, "Failed to initialize WifiStateMachine");
         }
         mWifiController.start();
+        mIsControllerStarted = true;
 
         // If we are already disabled (could be due to airplane mode), avoid changing persist
         // state here
@@ -817,6 +819,10 @@
             Binder.restoreCallingIdentity(ident);
         }
 
+        if (!mIsControllerStarted) {
+            Slog.e(TAG,"WifiController is not yet started, abort setWifiEnabled");
+            return false;
+        }
 
         if (mPermissionReviewRequired) {
             final int wiFiEnabledState = getWifiEnabledState();
